src/stream/src/executor/backfill/arrangement_backfill.rs

```diff
@@ -288,9 +290,9 @@ where
                                 match msg? {
                                     None => {
                                         // Consume remaining rows in the builder.
-                                        for (vnode, builder) in &mut builders {
+                                        for (vnode, (builder, limiter)) in &mut builders {
```

Comment:
Looks like a vnode level rate limit, I think we need 1 for the entire partition instead.